Add py37

ocefpaf opened the pull request.

ocefpaf committed:
update-cb3

ocefpaf committed:
MNT: Re-rendered with conda-smithy 3.1.10 and pinning 2018.07.24

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

ocefpaf commented:
All builds on Windows are failing :-(

```
running run_test.py
Traceback (most recent call last):
  File "C:\bld\netcdf4_1533495484405\test_tmp\run_test.py", line 5, in <module>
    import netCDF4
  File "C:\bld\netcdf4_1533495484405\_test_env\lib\site-packages\netCDF4\__init__.py", line 3, in <module>
    from ._netCDF4 import *
ImportError: DLL load failed: The specified module could not be found.
```

@isuruf do you have an idea of what mau be going on there? There is very little change since the last successful build :-/
